tools/rule_extractor/rule_extractor.py

```diff
@@ -0,0 +1,147 @@
+    )
+    parser.add_argument(
+        '-t',
+        '--type',
+        required=True,
+        dest='object_type',
+        help='Specify the type of objects to fetch',
+        choices=['rules', 'alerts']
+    )
+
+    args = parser.parse_args()
+    if not args.ip:
+        # Obtain VM ip by connecting to it and running a command
+        # Requires that the VM is up and running.
+        ip = subprocess.check_output(
+            [
+                "vagrant",
+                "ssh",
+                "--command",
+                "hostname -I | cut -d' ' -f1"
+            ]
+        ).decode(sys.stdout.encoding).strip()
+        # Validate that we got a real ipv4 address
+        try:
+            ipaddress.ip_address(ip)
+        except ValueError:
+            sys.stderr.write("Ip address: {} is invalid".format(ip))
+            sys.exit(1)
+    else:
+        ip = args.ip
+        try:
+            ipaddress.ip_address(ip)
+        except ValueError:
+            sys.stderr.write("Ip address: {} is invalid".format(ip))
+            sys.exit(1)
+
+    if args.object_type == "rules":
+        result = query_prometheus_api(ip=ip, port=args.port, route="rules")
+        # For documentation purposes, we don't want to litter the whole thing
+        # with all fields, so lets strip off some "ambiguous" fields
+        # Also we need to test that this generated default alerting rules are
+        # actually same as the deployed alerting rules
+        rule_group = result.get('data', {}).get('groups', [])
+        recording_rules = []
+        alerting_rules = []
+        for item in rule_group:
+            for rule in item.get('rules', []):
+                # rule type can be alerting or recording
+                # For now, we only need alerting rules
+                if rule['type'] == "alerting":
+                    message = rule['annotations'].get('message') or \
+                        rule['annotations'].get('summary')
+                    fixup_alerting_rule = {
+                        'name': rule['name'],
+                        'severity': rule['labels']['severity'],
+                        'message': message
+                    }
+                    alerting_rules.append(fixup_alerting_rule)
+
+                elif rule['type'] == "recording":
+                    fixup_record_rules = {
+                        'name': rule['name'],
+                        'query': rule['query']
+                    }
+                    recording_rules.append(fixup_record_rules)
+
+        # We do not need the recording rules at this point in time
+        # write_to_file(recording_rules, "recording_rules")
+        write_to_file(alerting_rules, "alerting_rules")
+    else:
+        # Returns a list of all active alerts if route <alerts> is queried
+        # we do not care about this for now
+        sys.stderr.write("Input argument {} is unsupported".format(
+            args.object_type)
+        )
+        sys.exit(1)
```

Comment:
Why having a mandatory option for something not implemented ?

Reply:
Will remove it